### DISCUSSION TITLE
About the `IEnumerator object is never disposed` warning

### DISCUSSION THREAD

**Discussion by stenci** [wtertinek/Linq2Acad#41] [Q&A] [answered]
I am exploring the library and noticed the warning `IEnumerator object is never disposed; consider safe-cast to IDisposable and Dispose() call after enumeration`:
![image](https://github.com/wtertinek/Linq2Acad/assets/5955495/f61ebb6a-fa3c-4a2c-bb5a-0bf70fc3c845)

This would be a solution:
```c#
    public override IEnumerator<T> GetEnumerator()
    {
      var enumerator = ids.GetEnumerator();
      try
      {
        while (enumerator.MoveNext())
        {
          yield return getID(enumerator.Current);
        }
      }
      finally
      {
        if (enumerator is IDisposable disposable)
        {
          disposable.Dispose();
        }
      }
    }
```

I would like to know if the warning was intentionally ignored, and if yes, why.

---

Reading your blog posts I learned something about working with AutoCAD APIs. I wish the official documentation had [this](https://wtertinek.com/2015/05/29/linq-and-the-autocad-net-api-part-3/#:~:text=get%20the%20following%3A-,TABLE%20TYPE,ViewTableId,-It%E2%80%99s%20important%20to) and [this](https://wtertinek.com/2015/05/31/linq-and-the-autocad-net-api-part-4/#:~:text=ID%20PROPERTY,DBVisualStyle) tables!

Exploring your library I am learning about clean C# and project organization.

Thank you!

**Comment by benkoshy** [comment]
> I would like to know if the warning was intentionally ignored, and if yes, why.

Isn't it automatically disposed in the `finally` block?

**Reply by stenci** [reply]
There is no `finally` block.

**Comment by wtertinek** [comment, marked as answer]
Hi, stenci,

I'm very glad my posts and the Code have been helpful. Feel free to reach out if you have any further questions.

I do not get the warning in my IDE (I'm using VS 2022).

Calling `Dispose()` is ignored here on purpuse, because the underlying object is an AutoCAD container object (eg. `BlockTable`) and it would have a negative effect if we dispose of the object here and then use it further down the call chain. Furthermore, there are only two AutoCAD classes I'm aware of whose objects have to be disposed of explicitly, `ObjectIdCollection` and `IdMapping`.

**Reply by stenci** [reply]
> I do not get the warning in my IDE (I'm using VS 2022).

I use JetBrains Rider 2022 and I have targeted .net 4.8 instead of 4.5. I guess one of the two caused the warning to appear.
